executor/union_scan_test.go

```diff
@@ -332,3 +332,28 @@ func (s *testSuite7) TestForUpdateUntouchedIndex(c *C) {
 	tk.MustExec("commit")
 	tk.MustExec("admin check table t")
 }
+
+func (s *testSuite7) TestUpdateScanningHandles(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("create table t(a int primary key, b int);")
+	tk.MustExec("begin")
+	for i := 2; i < 100000; i++ {
```

Comment:
This is a bench test? so much data may slow the CI.

Reply:
A correctness test. If the snapshot getter reads the newly added key, the result of this SQL will not empty. The first prototype of this PR doesn't implement snapshot getter, and it fails to pass this test.